dissect/target/helpers/configutil.py

```diff
@@ -69,29 +74,28 @@ def _collapse_dict(self, dictionary: dict, collapse: bool = False) -> dict[str,
             return dictionary
 
         for key, value in dictionary.items():
-            value = self._collapse_dict(value, self.collapse_all or key in self.collapse)
+            value = self._collapse_dict(value, self.collapse_all or self._collapse_check(key))
             new_dictionary.update({key: value})
 
         return new_dictionary
 
+    def _key_in_collapse(self, key: str) -> bool:
+        return key in self.collapse
+
+    def _key_not_in_collapse(self, key: str) -> bool:
+        return key not in self.collapse
+
```

Comment:
I would order these semi private `_` functions after the private `__` functions.